File: src/fast_food_restaurant.py
def calc_dishes(dishes):
  dishes = ''.join(sorted(map(lambda x: str(min(x, 4)), dishes)))

  if (dishes == "000"):
    return 0
  if (dishes.startswith("00")):
    return 1
  if (dishes.startswith("01")):
    return 2
  if (dishes.startswith("0") or dishes.startswith("11")):
    return 3
  if (dishes >= "122" and dishes <= "222"):
    return 4
  if (dishes >= "223" and dishes <= "244"):
    return 5
  if (dishes.startswith("3")):
    return 6
  return 7

rounds = int(input())

for i in range(rounds):
  a, b, c = map(int, input().split())
  print(calc_dishes([a, b, c]))

File: src/FastFoodRestaurant.java
import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class FastFoodRestaurant {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int rounds = scanner.nextInt();

        for (int i = 0; i < rounds; i++) {
            System.out.println(countMaxDishes(scanner.nextInt(), scanner.nextInt(), scanner.nextInt()));
        }
    }

    static int countMaxDishes(int a, int b, int c) {
        String entry = Stream.of(a, b, c)
                .map(n -> Math.min(n, 4))
                .map(String::valueOf)
                .sorted()
                .collect(Collectors.joining());
        int entryInt = Integer.parseInt(entry);

        if (entry.equals("000")) return 0;
        if (entry.startsWith("00")) return 1;
        if (entry.startsWith("01")) return 2;
        if (entry.startsWith("0") || entry.startsWith("11")) return 3;
        if (entryInt >= 122 && entryInt <= 222) return 4;
        if (entryInt >= 223 && entryInt <= 244) return 5;
        if (entry.startsWith("3")) return 6;
        return 7;
    }
}
